### PR TITLE
Feature/5 delete diary

### PR DESCRIPTION
**Pull request by onpyeong**

### CODE REVIEW

File: app/src/main/java/com/example/mydiary/DiaryAdapter.kt [hunk 1]
```diff
@@ -18,12 +28,19 @@ class DiaryAdapter : RecyclerView.Adapter<DiaryAdapter.DiaryItemViewHolder>() {
...
+            // MainActivity에서의 로직으로 set된 listener를 invoke 시켜서
+            // 현재 선택된 item을 delete함
+            deleteButtonClickListener?.invoke(diaryList[position])
+            notifyDataSetChanged()
+        }
     }
 
     override fun getItemCount(): Int = diaryList.size
 
     class DiaryItemViewHolder(
```
Comment (onpyeong) [inline app/src/main/java/com/example/mydiary/DiaryAdapter.kt:42]: inner class로 만드는 방법도 있음

File: app/src/main/java/com/example/mydiary/MainActivity.kt [hunk 2]
```diff
@@ -37,7 +49,7 @@ class MainActivity : AppCompatActivity() {
             it ->
             //update UI
             // observer 내부 onChanged 메소드에서 관찰하던 LiveData가 변하면 무엇을 할 것인지 액션 지정
-            diaryAdapter.diaryList.addAll(it)
+            diaryAdapter.diaryList = it as MutableList<DiaryData>
```
Comment (onpyeong) [inline app/src/main/java/com/example/mydiary/MainActivity.kt:52]: as 대신 .toMutableListOf()로 쓰기

Comment (onpyeong) [reply app/src/main/java/com/example/mydiary/MainActivity.kt:52]: clear 하고 addAll 할 수도 있음

File: app/src/main/java/com/example/mydiary/MainActivity.kt [hunk 3]
```diff
@@ -37,7 +49,7 @@ class MainActivity : AppCompatActivity() {
...
             diaryAdapter.notifyDataSetChanged()
```
Comment (mdb1217) [inline app/src/main/java/com/example/mydiary/MainActivity.kt:53]: DiffUtil 알아보기 ^~^

File: app/src/main/java/com/example/mydiary/DiaryAdapter.kt [hunk 4]
```diff
@@ -18,12 +28,19 @@ class DiaryAdapter : RecyclerView.Adapter<DiaryAdapter.DiaryItemViewHolder>() {
 
     override fun onBindViewHolder(holder: DiaryItemViewHolder, position: Int) {
         holder.onBind(diaryList[position])
+        holder.binding.btnClear.setOnClickListener {
+            Log.d("button", "click")
+            // MainActivity에서의 로직으로 set된 listener를 invoke 시켜서
+            // 현재 선택된 item을 delete함
+            deleteButtonClickListener?.invoke(diaryList[position])
+            notifyDataSetChanged()
```
Comment (onpyeong) [inline app/src/main/java/com/example/mydiary/DiaryAdapter.kt:36]: https://github.com/bbangmap/BBangMap-Android/pull/283/files/573f1ed3e530ab934e7742191c5b03655cd2482e#diff-6d38a6621ed11eb395a86e41b3519273e5dab6acf5180c79303183d73db52c50R86-R88:~:text=.startMultiImage%20%7B%20uriList%20%2D%3E,newReviewWriteImageAdapter.submitList(uriList)

File: app/src/main/java/com/example/mydiary/SwipeHelperCallback.kt [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    private lateinit var binding: ActivityMainBinding
+
+    override fun getMovementFlags(
+        recyclerView: RecyclerView,
+        viewHolder: RecyclerView.ViewHolder
+    ): Int {
+        // Drag, Swipe 방향 결정
+        return makeMovementFlags(0, LEFT or RIGHT)
+    }
+
```
Comment (onpyeong) [inline app/src/main/java/com/example/mydiary/SwipeHelperCallback.kt:22]: override fun getMovementFlags(
        recyclerView: RecyclerView,
        viewHolder: RecyclerView.ViewHolder
    ): Int {
        // Drag, Swipe 방향 결정
        return makeMovementFlags(0, LEFT or RIGHT)
    }
https://github.com/onpyeong/MyDiary/pull/6/files#diff-75460e114282c42c7bea7f0e7d0ae4c9939f4eaad6d866d264eda29103c81448R15-R22:~:text=binding%3A%20ActivityMainBinding-,override%20fun%20getMovementFlags(,%7D,-Commenting%20on%20lines

Comment (onpyeong) [reply app/src/main/java/com/example/mydiary/SwipeHelperCallback.kt:22]: https://github.com/onpyeong/MyDiary/pull/6/files#diff-75460e114282c42c7bea7f0e7d0ae4c9939f4eaad6d866d264eda29103c81448R15-R22:~:text=binding%3A%20ActivityMainBinding-,override%20fun%20getMovementFlags(,%7D,-Commenting%20on%20lines

File: app/src/main/java/com/example/mydiary/DiaryAdapter.kt [hunk 6]
```diff
@@ -18,12 +28,19 @@ class DiaryAdapter : RecyclerView.Adapter<DiaryAdapter.DiaryItemViewHolder>() {
 
     override fun onBindViewHolder(holder: DiaryItemViewHolder, position: Int) {
         holder.onBind(diaryList[position])
+        holder.binding.btnClear.setOnClickListener {
+            Log.d("button", "click")
+            // MainActivity에서의 로직으로 set된 listener를 invoke 시켜서
+            // 현재 선택된 item을 delete함
```
Comment (onpyeong) [inline app/src/main/java/com/example/mydiary/DiaryAdapter.kt:34]: [보여지는 텍스트](#https://github.com/onpyeong/MyDiary/pull/6/files#diff-44a2f936a16025127c48c55945b1aaf6c2ade29897952e3560587c5218c0f6abR31-R34:~:text=Log.d(,%EC%9D%84%20delete%ED%95%A8)

**Pull request merged**